### ISSUE TITLE
trafoXML's not backwards compatible anymore

### ISSUE THREAD

**Issue by cbielow** [OpenMS/OpenMS#6067]
after https://github.com/OpenMS/OpenMS/commit/ede172ce33db356bb4800035a109a2b34c424e54#diff-04addb6bc476771cea7865351524a610a6522f2e07babf6933af1fd64de2dbb2R63
and 
https://github.com/OpenMS/OpenMS/commit/333ebe745bd4b211f0f6ec851bc560db34a2d70d

a trafoXML generated before that change will yield
`weight  is not supported.Invalid parameter: Value '' is not a valid weight parameter for x values.` on the commandline when its being loaded.

This basically breaks all workflows which use old data.
Is there a reason that backwards compatibility was shut down so hard here?

**Comment by jpfeuffer** [comment]
No, you can add it back. As long as the empty string is not in the tool parameters.

**Comment by timosachsenberg** [comment]
I think it was because of optional vs empty parameters. Leads to problems in the galaxy wrappers iirc
